examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 The godruoyi/gno-snowflake Author. All rights reserved.
+// Use of this source code is governed by a MIT.
+
+package uuid
+
+import (
+	"errors"
+	"time"
+)
+
+// These constants are the bit lengths of snowflake ID parts.
+const (
+	TimestampLength uint8  = 41
+	MachineIDLength uint8  = 10
+	SequenceLength  uint8  = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint64 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+)
+
+// default start time is 2008-11-10 23:00:00 UTC, why ? In the playground the time begins at 2009-11-10 23:00:00 UTC.
+// It can run on golang playground.
+// default machineID is 0
+type SequenceResolver func(ms int64) (uint16, error)
+
+var (
+	resolver      SequenceResolver
+	MachineID     uint64 = 0
+	sequence      uint16 = 0
+	StartTime            = time.Date(2008, 11, 10, 23, 0, 0, 0, time.UTC)
+	LastTimestamp int64  = 0
+	// _             int    //will not sync atomic Gno not goroutine, implement in the future for the resolver.
+)
+
+// ID use ID to generate snowflake id, and it will ignore error. if you want error info, you need use NextID method.
+func ID() uint64 {
```

Comment:
What are the advantages of using an `uint64` over an `int`, here?

I think `int` is more appropriate.

Reply:
`uint64` is used for positive values ranging from `0 to 2^64-1` and `int` is used for values ranging from `-2^63 to 2^63-1`
I think using `uint64` is better because `uuid` is a positive value and `uint64` has a larger range of positive values. If we use `int`, we would only have half of the values since we do not want negative values, and this could create errors with those values